### PR TITLE
Create movement boundaries to 3 spaces

### PR DESCRIPTION
**Pull request by 0xNonCents**
## In this PR
- Prevents saving of a move if it is not within 3 spaces
- Highlight the area that is within 3 spaces
## Future PR
Only show highlighting when a squad that you own is selected

### CODE REVIEW

File: react-app/src/ui/components/three/HexagonBackground.tsx [hunk 1]
```diff
@@ -67,6 +69,17 @@ export const HexagonBackground = ({
...
+  //Any comments on how I am using dojo to read squad positions is appreciated.
+  const positions = useEntityQuery([Has(Position)]).map((entity) => {
+    const positions = useComponentValue(Position, entity);
+    return positions;
+  });
+  //See of selected hex is on a squad position
+  const isOnSquad = positions.filter(p => p?.game_id == 5).some((pos) => {
+    pos?.x === selectedHex?.col && pos?.y === selectedHex?.row;
+  });
+
```
Comment (0xNonCents) [inline react-app/src/ui/components/three/HexagonBackground.tsx]: isOnSquad is not used atm so selecting anywhere shows the bounds. What is the origin coordinate for squads? I see x : 1000 and y : 1000 being around where most coordinates are.

Screenshot 2024-02-06 at 12 40 09 PM

Comment (ponderingdemocritus) [reply react-app/src/ui/components/three/HexagonBackground.tsx]: the first squad spawns at 1000,1000. We just offset in the client so we don't underflow in cairo

File: react-app/src/ui/components/three/HexagonBackground.tsx [hunk 2]
```diff
@@ -67,6 +69,17 @@ export const HexagonBackground = ({
...
   } = useStateStore();
 
+
+  //Any comments on how I am using dojo to read squad positions is appreciated.
+  const positions = useEntityQuery([Has(Position)]).map((entity) => {
+    const positions = useComponentValue(Position, entity);
+    return positions;
+  });
+  //See of selected hex is on a squad position
+  const isOnSquad = positions.filter(p => p?.game_id == 5).some((pos) => {
```
Comment (ponderingdemocritus) [inline react-app/src/ui/components/three/HexagonBackground.tsx]: this already exists as a query 

```
  const squadsOnHex = useEntityQuery([
    Has(Position),
    HasValue(Position, { x: col + offset, y: row + offset }),
  ]);
  ```

File: react-app/src/ui/components/three/HexagonBackground.tsx (hunk 1 above)
Comment (0xNonCents) [reply react-app/src/ui/components/three/HexagonBackground.tsx]: I see offset as 
`export const offset = 995;`
is this right?

**Pull request closed without merging**